### PR TITLE
Update tests

### PR DESCRIPTION
**Pull request by drwells**
We have a bunch of tests which fail for trivial reasons - I fixed a bunch here.

### CODE REVIEW

File: tests/hp/non_hp_mode.cc [hunk 1]
```diff
@@ -47,7 +47,8 @@ test()
     }
   catch (ExceptionBase &e)
     {
-      deallog << e.what() << std::endl;
+      deallog << "Successfully failed to distribute DoFs with a single "
```
Comment (bangerth) [inline tests/hp/non_hp_mode.cc:50]: "successfully failed" :-)

File: tests/hp/non_hp_mode.cc [hunk 2]
```diff
@@ -47,7 +47,8 @@ test()
...
+              << "nonzero active FE index" << std::endl;
```
Comment (bangerth) [inline tests/hp/non_hp_mode.cc:51]: You might want to comment out the name of `e` here as well.

File: tests/sparsity/sparsity_pattern_base_01.cc [hunk 3]
```diff
@@ -56,7 +56,11 @@ main()
     entries;
 
   for (unsigned int i = 0; i < 100; ++i)
-    entries.emplace_back(rand() % 10, rand() % 100);
+    {
+      const auto row = Testing::rand() % 10;
+      const auto col = Testing::rand() % 100;
+      entries.emplace_back(row, col);
```
Comment (bangerth) [inline tests/sparsity/sparsity_pattern_base_01.cc:62]: It took me a minute to see the difference -- nice catch!

**Review by bangerth — APPROVED** [review]

**Comment by stefanozampini** [comment]
@drwells You will need to add your changes to the new tests I have here https://github.com/dealii/dealii/pull/15276 too.

**Comment by drwells** [comment]
The PETSc TS tests now fail to run for me (I'm using a recent development copy of PETSc). For now I just removed them from this PR.

**Review by masterleinad — APPROVED** [review]

**Comment by stefanozampini** [comment]
> The PETSc TS tests now fail to run for me (I'm using a recent development copy of PETSc). For now I just removed them from this PR.

What do you mean fail to run? Before removing tests it could be good to know why they are failing

**Comment by drwells** [comment]
For `petsc_ts_04.debug` I now see
```
11599: DEAL::Evaluating explicit function at t=0.100000
11599: DEAL::Evaluating explicit function at t=0.100000
11599: DEAL::Evaluating explicit function at t=0.100000
11599: DEAL::Evaluating explicit function at t=0.100000
11599: DEAL::Evaluating explicit function at t=0.200000
11599: DEAL::Evaluating explicit function at t=0.200000
11599: DEAL::Evaluating explicit function at t=0.200000
11599: DEAL::Evaluating explicit function at t=0.200000
11599: DEAL::Intermediate output:
11599: DEAL::  t =0.200000
11599: DEAL::  y =0.823864  (exact: 0.818731)
11599: DEAL::Evaluating explicit function at t=0.400000
11599: DEAL::Time step too large: last_eval_time=0.200000, t=0.400000
11599: 
11599: petsc/petsc_ts_04.debug: RUN failed. ------ Additional output on stdout/stderr:
11599: 
11599: 
11599: 
11599: petsc/petsc_ts_04.debug: ******    RUN failed    *******
11599: 
11599: ===============================    OUTPUT END   ===============================
11599: Expected stage PASSED - aborting
11599: CMake Error at /not_backed_up/drwells/Documents/Code/CPP/dealii/build/share/deal.II/scripts/run_test.cmake:116 (message):
11599:   *** abort
```

**Comment by stefanozampini** [comment]
> For `petsc_ts_04.debug` I now see
> 
> ```
> 11599: DEAL::Evaluating explicit function at t=0.100000
> 11599: DEAL::Evaluating explicit function at t=0.100000
> 11599: DEAL::Evaluating explicit function at t=0.100000
> 11599: DEAL::Evaluating explicit function at t=0.100000
> 11599: DEAL::Evaluating explicit function at t=0.200000
> 11599: DEAL::Evaluating explicit function at t=0.200000
> 11599: DEAL::Evaluating explicit function at t=0.200000
> 11599: DEAL::Evaluating explicit function at t=0.200000
> 11599: DEAL::Intermediate output:
> 11599: DEAL::  t =0.200000
> 11599: DEAL::  y =0.823864  (exact: 0.818731)
> 11599: DEAL::Evaluating explicit function at t=0.400000
> 11599: DEAL::Time step too large: last_eval_time=0.200000, t=0.400000
> 11599: 
> 11599: petsc/petsc_ts_04.debug: RUN failed. ------ Additional output on stdout/stderr:
> 11599: 
> 11599: 
> 11599: 
> 11599: petsc/petsc_ts_04.debug: ******    RUN failed    *******
> 11599: 
> 11599: ===============================    OUTPUT END   ===============================
> 11599: Expected stage PASSED - aborting
> 11599: CMake Error at /not_backed_up/drwells/Documents/Code/CPP/dealii/build/share/deal.II/scripts/run_test.cmake:116 (message):
> 11599:   *** abort
> ```

@drwells I don't understand the reasons for the failures. May I ask you to provide the output of the tests with the following options?

`-ts_view -snes_linesearch_monitor -snes_monitor -ts_monitor -ts_adapt_monitor -ksp_converged_reason`

You can add them in an environment variable `PETSC_OPTIONS`

**Comment by bangerth** [comment]
This has tested cleanly, and I think it is clearly a positive step, so I'm going to merge. 

@drwells Can you open a report about whatever you think is still open regarding the comments you have above?

**Pull request merged**